spa/src/oauth-manager/index.tsx

```diff
@@ -45,6 +47,23 @@ const OauthManager = () => {
 		return false;
 	}
 
+	async function installNewApp(onFinish: () => void) {
+		const app = await ApiRequest.gitHubApp.getAppNewInstallationUrl();
+		const exp = new Date(new Date().getTime() + FIFTEEN_MINUTES_IN_MS);
+		document.cookie = `is-spa=true; expires=${exp.toUTCString()}; path=/; SameSite=None; Secure`;
+		const winInstall = window.open(app.data.appInstallationUrl, "_blank");
+		const hdlWinInstall = setInterval(() => {
```

Comment:
I think its better to use the `postMessage` similar to the how its used in `ConfigSteps.tsx`?
Simply pass a message after closing the child window and grab that message here and implement this method.

Reply:
Yup, will try to fix that.

Reply:
Can we merged this one for now? I need to think about how to structure the code. Don't want to add duplicate events listener for the clearing.